src/main/webapp/app/shared/markdown-editor/commands/exerciseReference.command.ts

```diff
@@ -0,0 +1,27 @@
+import { MultiOptionCommand } from 'app/shared/markdown-editor/commands/multiOptionCommand';
+import { MetisService } from 'app/shared/metis/metis.service';
+
+export class ExerciseReferenceCommand extends MultiOptionCommand {
+    metisService: MetisService;
+
+    buttonTranslationString = 'artemisApp.metis.editor.exercise';
+
+    constructor(metisService: MetisService) {
+        super();
+        this.metisService = metisService;
+    }
+
+    /**
+     * @function execute
+     * @param {string} selectedExerciseId   ID of the exercise to be referenced
+     * @desc                                Add an exercise reference link in markdown language
+     *                                      1. Add '[{exercise-title}](/courses/{courseId}/exercises/{exerciseId}})' at the cursor in the editor
+     *                                      2. Link in markdown language appears which when clicked navigates to the exercise page
+     */
+    execute(selectedExerciseId: string): void {
+        const selectedExercise = this.getValues().find((value) => value.id.toString() === selectedExerciseId);
+        const referenceLink = '[' + selectedExercise!.value + '](' + this.metisService.getLinkForExercise(selectedExercise!.id) + ')';
```

Comment:
Is there are case where `selectedExercise` has no value?

Reply:
There is none actually. We just have to access the object again to fetch some other parameters than the id and title, which were provided to the drop-down. I have moved the exclamation marks to the end of the `find`  function so that their purpose is more clear.